user-service/src/main/java/org/java/mentorship/user/repository/SessionRepository.java

```diff
@@ -0,0 +1,30 @@
+package org.java.mentorship.user.repository;
+
+import org.apache.ibatis.annotations.Insert;
+import org.apache.ibatis.annotations.Mapper;
+import org.apache.ibatis.annotations.Options;
+import org.apache.ibatis.annotations.Select;
+import org.java.mentorship.contracts.user.dto.Session;
+
+import java.util.List;
+import java.util.Optional;
+
+@Mapper
+public interface SessionRepository {
+    @Select("<script>" +
+            "SELECT * FROM sessions " +
```

Comment:
Two points here:
1. You can use a multi-line string with proper indentation
```
@Select("""
            <script>
            SELECT * FROM sessions
                <where> 
                <if test='userId != null'>
                AND user_id=#{userId}
                </if>
                </where>
            </script> 
            """)
```

2. If userId is not null, I think the SQL will have a syntax error, since it will be
SELECT * FROM sessions WHERE AND user_id=#{userId}

Reply:
This does not happen, the <where> tag of Mybatis makes sure there are no extra operators before and after inside it